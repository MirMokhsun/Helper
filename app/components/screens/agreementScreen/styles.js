import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    scroll: {
        paddingTop: 20
    },
    wrapper: {
        flex: 1,
        width: "100%",
        height: "100%",
        backgroundColor: '#F5F5F7',
    },
    regularText: {
        textAlign: 'left',
        color: '#000756',
        fontSize: 12,
        fontFamily: 'Roboto-Light',
        lineHeight: 16,
        marginBottom: 15,
        marginRight: 10,
        marginLeft: 10,
    },
    titleText: {
        textAlign: 'left',
        color: '#1459D2',
        fontSize: 16,
        fontFamily: 'Roboto-Light',
        lineHeight: 18,
        marginBottom: 15,
        marginRight: 10,
        marginLeft: 10,
    }, 
    buttonWrapper:{
        marginVertical:35,
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    }
})