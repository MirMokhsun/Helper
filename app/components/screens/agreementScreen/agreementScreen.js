import React from 'react';
import { View, Text, FlatList, StatusBar } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { agreementDoc } from './textAgreement';
import { styles } from './styles';
import { localization } from '../../../common/localization';
import ButtonSubmit from '../../commonComponents/buttonSubmit/buttonSubmit';
import { setRegistrationCheckbox } from '../registrationScreen/redux/registrationDataAction';

export class AgreementScreen extends React.Component {
    static navigationOptions = ({ navigation }) => {
        return {
            title: navigation.state.params ? navigation.state.params.title ? navigation.state.params.title : 'User agreement' : 'User agreement',
            headerRight: (<View />),
        }
    };

    constructor(props) {
        super(props)
        this.state = {
            allAgreementText: null
        }
    }

    getText = () => {
        let agreemet = null;
        const { language } = this.props;
        const objectsAgreemet = this.returnAgreementDocWithLocalization(language);
        if (objectsAgreemet) {
            agreemet = [];
            for (const key in objectsAgreemet) {
                let textStyle = null;
                if (objectsAgreemet[key].textstyle === 'regularText') {
                    textStyle = styles.regularText;
                } else {
                    textStyle = styles.titleText;
                }
                agreemet.push(
                    <Text
                        key={key}
                        style={textStyle}
                    >{objectsAgreemet[key].text}</Text>
                );
            }
            agreemet.push(
                <View style={styles.buttonWrapper}>
                    <ButtonSubmit
                        text={localization[language].introduced}
                        testID="ButtonAgreementID"
                        isDisabled={false}
                        isShowSpiner={false}
                        onPress={this.backToRegistration}
                    />
                </View>
            );
        }
        return agreemet;
    }

    backToRegistration = () => {
        const { agreementCheckbox, setRegistrationCheckbox, navigation } = this.props;
        if (!agreementCheckbox) {
            setRegistrationCheckbox();
        }
        navigation.goBack();
    }

    returnAgreementDocWithLocalization = (languageName) => {
        let agreement = agreementDoc.en;
        switch (languageName) {
            case 'ru':
                agreement = agreementDoc.ru;
                break;
            case 'es':
                agreement = agreementDoc.es;
                break;
            case 'ar':
                agreement = agreementDoc.ar;
                break;
            default:
                break;
        }
        return agreement
    }

    componentDidMount() {
        const allAgreementText = this.getText();
        this.setState({
            allAgreementText
        });
        const { navigation } = this.props;
        const { title } = navigation.state.params;
        navigation.setParams({ title });
    }

    render() {
        const { allAgreementText } = this.state;
        return (
            <View style={styles.wrapper}>
                <StatusBar barStyle="light-content" backgroundColor="#000756" />
                <FlatList
                    data={allAgreementText}
                    renderItem={({ item }) => item}
                    keyboardShouldPersistTaps="handled"
                    style={styles.scroll}
                    contentContainerStyle={{ backgroundColor: '#F5F5F7' }} />
            </View>
        )
    }
}


const mapStateToProps = state => ({
    language: state.currentLanguage.language,
    agreementCheckbox: state.registrationData.agreementCheckbox.isValid
});
const mapDispatchToProps = dispatch => bindActionCreators({
    setRegistrationCheckbox,
}, dispatch);
export default connect(mapStateToProps, mapDispatchToProps)(AgreementScreen);