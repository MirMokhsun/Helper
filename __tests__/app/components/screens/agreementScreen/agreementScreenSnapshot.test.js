import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AgreementScreen } from '../../../../../app/components/screens/agreementScreen/agreementScreen';

it('create snapshot AgreementScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<AgreementScreen
        setRegistrationCheckbox={() => { }}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});