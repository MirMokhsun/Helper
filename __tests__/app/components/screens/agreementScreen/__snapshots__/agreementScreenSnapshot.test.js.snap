// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AgreementScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F7",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#000756"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <FlatList
    contentContainerStyle={
      Object {
        "backgroundColor": "#F5F5F7",
      }
    }
    data={null}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "paddingTop": 20,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
